src/resources/games/controller.js

```diff
@@ -6,10 +6,148 @@ async function getAllGames(req, res) {
 
     res.status(200).json({ games });
   } catch (error) {
-    console.error("[ERROR] /signup route: ", error);
+    console.error("[ERROR] /games route: ", error);
 
     res.status(500).json({ error });
   }
 }
 
-module.exports = { getAllGames };
+async function getOneGame(req, res) {
+  const targetId = parseInt(req.params.id);
+  try {
+    const game = await prisma.game.findFirst({
+      where: {
+        id: targetId,
+      },
+      include: {
+        users: {
+          include: {
+            user: {
+              select: {
+                id: true,
+                username: true,
+              },
+            },
+          },
+        },
+      },
+    });
+
+    const cleanResult = {
+      ...game,
+      users: game.users.map((user) => {
+        return user.user;
+      }),
+    };
+
+    res.status(200).json({ cleanResult });
+  } catch (error) {
+    console.error(`[ERROR] /games/${targetId} route: `, error);
+
+    res.status(500).json({ error });
+  }
+}
+
+async function joinGame(req, res) {
+  const targetId = parseInt(req.params.id);
+
+  const userId = req.body.userId;
+
+  try {
+    const result = await prisma.game.update({
+      /* Update doesn't allow multiple where conditions. 
+      Had to create extra condition `gameStatusId` in game table.
+      Refecence: // https://github.com/prisma/prisma/discussions/4185
+      */
+      where: {
+        gameStatusId: {
+          id: targetId,
+          gameStatus: "waiting",
+        },
```

Comment:
What's the reason you aren't doing:

```
where: {
  id: targetId
}
```

The id in `req.params` is the if for a game correct? Not a user?

Reply:
Yes, `req.params` is for the game id. The reason why I'm doing:
```
where: {
        gameStatusId: {
          id: targetId,
          gameStatus: "waiting",
        }
```
(as I described in the PR description) is that I want to control new `gameStatus` depending on the old one, eg. when the user creates a game, gameStatus should be "waiting", when other user joins that game, it should change to "in-progress" and the game begins. In this case I can also controll that there will be just two players in this specific game.